src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotAlwaysSkipTokenValidationInDelegates.cs

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotAlwaysSkipTokenValidationInDelegates : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA5405"; // TODO: Somewhere in documentation I saw I need to create a CA5405.md file, don't recall where.
+        private static readonly LocalizableString s_Title = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.DoNotAlwaysSkipTokenValidationInDelegates),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.DoNotAlwaysSkipTokenValidationInDelegatesMessage),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.DoNotAlwaysSkipTokenValidationInDelegatesDescription),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                RuleLevel.BuildWarning,
+                description: s_Description,
+                isPortedFxCopRule: false,
+                isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
```

Comment:
Can you rename this to `context`? (Intentionally reusing and shadowing the existing name)

Reply:
What benefit does this offer (and below)? I'm following a pattern from another analyzer here and it seems to help legibility to make it obvious that these variables belong to different scopes.

Reply:
@TimHannMSFT One of the benefits is that this prevents accidentally using a cancellation token from an outer context. I don't know if there are other benefits. But @sharwell will know.